### PR TITLE
Allow specification of chef versions

### PR DESCRIPTION
**Pull request by kamidzi**
Corollary to unfiled fix for #953

### CODE REVIEW

File: bootstrap/shared/shared_configure_chef.sh [hunk 1]
```diff
@@ -70,7 +83,11 @@ do_on_node bootstrap "$KNIFE cookbook upload -a \
 # install and bootstrap Chef on cluster nodes
 i=1
 for vm in $vms $mon_vms; do
-  do_on_node $vm "sudo dpkg -i \$(find $FILECACHE_MOUNT_POINT/ -name chef_\*deb -not -name \*downloaded | tail -1)"
+  # Try to install a specific version, or just the latest
+  if [[ -z "$CHEF_CLIENT_DEB" ]]; then
+    echo "Installing latest chef-client found in $vm:$FILECACHE_MOUNT_POINT"
+  fi
+  do_on_node bootstrap "$CHEF_CLIENT_INSTALL_CMD"
```
Comment (erhudy) [inline bootstrap/shared/shared_configure_chef.sh]: This needs to be `do_on_node $vm` rather than `do_on_node bootstrap`.

**Comment by erhudy** [comment]
After changing the `do_on_node bootstrap` line marked above to `do_on_node $vm` (so that the Chef client is upgraded on each VM instead of the unintentional effect of upgrading the client repeatedly on the bootstrap node), this works as expected.

**Comment by erhudy** [comment]
Re-checking and will merge on success.

**Comment by erhudy** [comment]
One observation before merge: the if statement at https://github.com/bloomberg/chef-bcpc/pull/961/files#diff-d37421a6d1f4a8b5352df28c5d878870R87 never seems to be executed, even if you don't set `CHEF_CLIENT_DEB` anywhere. I think this behavior is still fine because you will get the default version of Chef that we want you to use if you are not explicitly specifying one, just seems slightly different than intended.

**Comment by kamidzi** [comment]
I didn't run into that; the idea is that it is initialized at https://github.com/bloomberg/chef-bcpc/blob/f9a36f9e94ef3e99da2c55cd7bab19f65c551113/bootstrap/config/bootstrap_config.sh.defaults#L76. We may want to fix this if that's the case.

**Pull request merged**